src/common/util.c

```diff
@@ -136,11 +136,12 @@ util_is_zeroed(const void *addr, size_t len)
  * If insert is true, the calculated checksum is inserted into
  * the range at *csump.  Otherwise the calculated checksum is
  * checked against *csump and the result returned (true means
- * the range checksummed correctly).
+ * the range checksummed correctly). If nocheck is true, the new
+ * checksum is returned without check.
  */
-int
+uint64_t
 util_checksum(void *addr, size_t len, uint64_t *csump,
-	int insert, size_t skip_off)
+	int insert, size_t skip_off, size_t nocheck)
```

Comment:
Controlling the type of return value by a parameter is ugly.
I think this can be fixed by extracting the part you want to a new function - util_checksum_compute.